kaggle_environments/agent.py

```diff
@@ -84,42 +85,60 @@ def runner(raw, message):
 
 class Agent():
 
-    def __init__(self, raw, configuration, id=None):
+    def __init__(self, raw, configuration, use_process, id=None):
         self.id = id or str(uuid.uuid1())
-        self.manager = Manager()
-        self.message = self.manager.Namespace()
-        self.message.action = None
-        self.message.state = None
-        self.message.timeout = 0
-        self.message.configuration = configuration
-        self.process = Process(target=runner, args=(raw, self.message))
-        self.process.daemon = True
-        self.process.start()
+        self.use_process = use_process
+        if use_process:
+            self.manager = Manager()
+            self.message = self.manager.Namespace()
+            self.message.action = None
+            self.message.state = None
+            self.message.timeout = 0
+            self.message.configuration = configuration
+            self.process = Process(target=runner, args=(raw, self.message))
+            self.process.daemon = True
+            self.process.start()
+        else:
+            self.configuration = configuration
+            self.agent = build_agent(raw)
```

Comment:
self.agent = None - you don't want to run it yet (this will run the user's code immediately if a file or raw python string)